ttnn/cpp/pybind11/pytensor.cpp

```diff
@@ -1555,6 +1581,20 @@ void pytensor_module(py::module& m_tensor) {
             py::return_value_policy::reference)
         .def(
             "to_torch",
+            [](const Tensor& self) -> py::object { return detail::convert_tt_tensor_to_torch_tensor(self, true); },
+            R"doc(
+            Convert tensor to torch tensor using legacy padded shape.
+            WARNING: Will be deprecated soon!
+
+            The tensor must be on host when calling this function.
+
+            .. code-block:: python
+
+                data = tt_tensor.cpu().to_torch() # move TT Tensor to host and convert it to torch tensor
+
+        )doc")
+        .def(
+            "to_torch_with_logical_shape",
```

Comment:
this will be the new name going forward? I don't like this.

Reply:
This should be `to_torch()`, but problem is, I couldn't get rid of/replace `to_torch()` because there are code that depends on `to_torch()` to return padded tensor. There is a way forward to clean this up:
- Get rid of all `to_torch()`/`to_torch_with_logical_shape` and replace with functional version: `ttnn.from_torch()`
- Update code in `ttnn.from_torch()` to not use/depend on `to_torch()` behaviour
- Remove `to_torch()` and rename `to_torch_with_logical_shape()` to `to_torch()` (or just bind it to functional version and have same underlying functionality

I already integrated `to_torch_with_logical_shape()` into `ttnn.from_torch()` and `ttnn.to_torch()` and I will broadcast this as the API to use. The name of `to_torch_with_logical_shape()` doesn't really matter in this sense.